src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -335,7 +336,7 @@ public ClientBuilder get() {
 
   private final URI uri;
   private final String apiVersion;
-  private final RegistryAuth registryAuth;
+  private RegistryAuthSupplier registryAuthSupplier;
```

Comment:
you just need to check the `builder.registryAuthSupplier` against null, not `this.registryAuthSupplier`